File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        int[] po = new int[10];
        Queue<Process> que = new ArrayDeque<>();
        int maxPo = 0;
        
        for(int i=0; i<priorities.length; i++){
            int p = priorities[i];
            po[p]++;
            maxPo = Math.max(maxPo, p);
            que.add(new Process(i, p));
        }
        
        while(!que.isEmpty()){
            Process p = que.poll();
            if(p.priority < maxPo){
                que.add(p);
                continue;
            }
            answer++;
            po[p.priority]--;
            if(po[p.priority] == 0){
                for(int i=9; i>=0; i--){
                    if(po[i]>0){
                        maxPo = i;
                        break;
                    }
                }
            }
            //System.out.println(p.idx + " : " + p.priority);
            if(p.idx == location){
                break;
            }
        }
        
        return answer;
    }
}

class Process{
    int idx;
    int priority;
    
    public Process(int idx, int priority){
        this.idx = idx;
        this.priority = priority;
        
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
class process:
    def __init__(self, idx, prior):
        self.idx = idx
        self.prior = prior
        
    def __str__(self):
        return str(self.idx) + ' ' + str(self.prior)

def solution(priorities, location):
    
    que = []
    
    answer = 0
    
    for i in range(len(priorities)):
        que.append(process(i, priorities[i]))
    
    while len(que) > 0:
        
        p = que.pop(0)
        
        if p.prior == max(priorities):
            answer += 1
            if p.idx == location:
                break
            priorities.remove(p.prior)
        else:
            que.append(p)
            
        
    
    return answer